src/overlays/actors/ovl_En_Po_Sisters/z_en_po_sisters.c

```diff
@@ -192,673 +218,700 @@ void EnPoSisters_Destroy(Actor* thisx, PlayState* play) {
+        this->poSisterFlags |= (POSISTERS_FLAG_UPDATE_BGCHECK_INFO | POSISTERS_FLAG_MATCH_PLAYER_HEIGHT);
+        if (this->type == POSISTER_TYPE_MEG) {
+            EnPoSisters_MegCloneVanish(this, play);
         } else {
-            func_80B1AC40(this);
+            EnPoSisters_SetupAimlessIdleFlying(this);
         }
     } else {
-        f32 temp = this->skelAnime.curFrame / this->skelAnime.endFrame;
-        s32 temp_f16 = 255.0f * temp;
+        f32 alphaPercent = this->skelAnime.curFrame / this->skelAnime.endFrame;
+        s32 alpha = 255.0f * alphaPercent;
 
-        this->unk_229 = CLAMP(temp_f16, 0, 255);
+        this->color.a = CLAMP(alpha, 0, 255);
     }
 }
 
-void func_80B1C408(EnPoSisters* this, PlayState* play) {
-    Vec3f sp3C;
+void EnPoSisters_CheckCollision(EnPoSisters* this, PlayState* play) {
+    Vec3f pos;
 
     if (this->collider.base.acFlags & AC_HIT) {
         this->collider.base.acFlags &= ~AC_HIT;
         Actor_SetDropFlag(&this->actor, &this->collider.info);
 
-        if (this->unk_18D != 0) {
-            ((EnPoSisters*)this->actor.parent)->unk_194--;
+        if (this->megCloneId != POSISTER_MEG_REAL) {
+            ((EnPoSisters*)this->actor.parent)->megAttackTimer--;
             Actor_PlaySfxAtPos(&this->actor, NA_SE_EN_PO_LAUGH2);
-            func_80B1BE4C(this, play);
+            EnPoSisters_MegCloneVanish(this, play);
             if (Rand_ZeroOne() < 0.2f) {
-                sp3C.x = this->actor.world.pos.x;
-                sp3C.y = this->actor.world.pos.y;
-                sp3C.z = this->actor.world.pos.z;
-                Item_DropCollectible(play, &sp3C, ITEM00_ARROWS_10);
+                pos.x = this->actor.world.pos.x;
+                pos.y = this->actor.world.pos.y;
+                pos.z = this->actor.world.pos.z;
+                Item_DropCollectible(play, &pos, ITEM00_ARROWS_10);
             }
         } else if (this->collider.base.colType != 9) {
-            if (this->actor.colChkInfo.damageEffect == 0xF) {
+            if (this->actor.colChkInfo.damageEffect == POSISTERS_DAMAGEEFFECT_DEKUNUT) {
                 this->actor.world.rot.y = this->actor.shape.rot.y;
-                this->unk_191 |= 2;
-                func_80B1B860(this, play);
-            } else if ((this->unk_18C == 0) && (this->actor.colChkInfo.damageEffect == 0xE) &&
-                       (this->actionFunc == func_80B1C0A4)) {
-                if (this->unk_194 == 0) {
-                    this->unk_194 = -45;
+                this->poSisterFlags |= POSISTERS_FLAG_UPDATE_SHAPE_ROT;
+                EnPoSisters_SetupSpinBackToVisible(this, play);
+            } else if ((this->type == POSISTER_TYPE_MEG) &&
+                       (this->actor.colChkInfo.damageEffect == POSISTERS_DAMAGEEFFECT_SPINATTACK) &&
+                       (this->actionFunc == EnPoSisters_MegSurroundPlayer)) {
+                if (this->megAttackTimer == 0) {
+                    this->megAttackTimer = -45;
```

Comment:
What's the meaning behind this timer being negative?

Reply:
I commented this in the union where this variable is described, but this timer is counting backwards for `megAttackTimer`: setting negative and then incrementing up to zero.

This is the period where Meg has stopped spinning around the player, and is staring at the player before spin attacking at them, for some reason that has a reverse timer.

Reply:
ah wait it might be more nuanced, need to double check